windows/src/desktop/kmshell/install/UfrmInstallKeyboardLanguage.pas

```diff
@@ -413,6 +415,25 @@ procedure TfrmInstallKeyboardLanguage.FillLanguageGrid;
     gridLanguages.Objects[0,n] := Item;
     Inc(n);
   end;
+
+  function AddAllowableTag(const tag: string): TCustomLanguage;
+  begin
+    with TBCP47Tag.Create(tag) do
+    try
+      // Let's lookup the lang - script - region and get a good name
+      Result.Tag := Tag;
+      if not TLanguageCodeUtils.BCP47Languages.TryGetValue(Language, Result.LanguageName) then
+        Result.LanguageName := '';
+      if not TLanguageCodeUtils.BCP47Scripts.TryGetValue(Script, Result.ScriptName) then
+        Result.ScriptName := '';
+      if not TLanguageCodeUtils.BCP47Regions.TryGetValue(Region, Result.RegionName) then
+        Result.RegionName := '';
+      Result.FullName := TLanguageCodeUtils.LanguageName(Result.LanguageName,
+        Result.ScriptName, '');
```

Comment:
You don't want to include the region name in the `FullName`?

Reply:
No, because the region name is shown elsewhere in the grid - see the screenshots above